extensions/test-processors/CMakeLists.txt

```diff
@@ -17,7 +17,7 @@
 # under the License.
 #
 
-if (SKIP_TESTS AND NOT ENABLE_TEST_PROCESSORS)
+if (SKIP_TESTS AND NOT (ENABLE_ALL OR ENABLE_TEST_PROCESSORS))
```

Comment:
do we want to add the test processors as well? if enabled, this extension is also added to the build artifact (the `register_extension` does not seem to discriminate), so either we should improve `register_extension` to not add this or hold off on enabling this with `ENABLE_ALL`

Reply:
My thought was that as these processors are only built if the `SKIP_TESTS` is `OFF`, which means we are building the tests that need this extension anyway. If in this case we do not specifically set `ENABLE_TEST_PROCESSORS` because we have already set `ENABLE_ALL` to `ON` then we probably should build the test processors as they will be needed for building the tests.

Reply:
we could tie it to just `SKIP_TESTS` as you mention this is needed anyway for the tests

Reply:
The problem with that is that these processors are also used in the docker tests where the SKIP_TESTS is always ON.

Reply:
building a package with `ENABLE_ALL` will now include the test processors which could be confusing

Reply:
That could be confusing, removed in 24d0b4b9e848d2eb77ee7b4f3aa4e0b62130e45f